### PR TITLE
New probe endpoints

### PR DESCRIPTION
**Pull request by odisseus**
* Rerun failed tests
 * Set the project compiler output

### CODE REVIEW

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/Projects.scala [hunk 1]
```diff
@@ -118,6 +119,12 @@ object Projects extends IntelliJApi {
       }
   }
 
+  def setCompilerOutput(projectRef: ProjectRef, path: Path): Unit = {
+    val project = Projects.resolve(projectRef)
+    val compilerProjectExtension = CompilerProjectExtension.getInstance(project)
+    compilerProjectExtension.setCompilerOutputUrl(path.toString)
```
Comment (lukaszwawrzyk) [inline probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/Projects.scala:125]: shouldn't you convert Path to URL and then toString?

File: extensions/scala/tests/src/test/scala/org/virtuslab/ideprobe/scala/ScalaTestRunConfigurationTest.scala [hunk 2]
```diff
@@ -24,7 +24,13 @@ class ScalaTestRunConfigurationTest extends ScalaPluginTestSuite {
 
     val moduleRunConfiguration = ScalaTestRunConfiguration.Module(moduleRef)
     val moduleRunResult = intelliJ.probe.run(moduleRunConfiguration)
+    val allTests = moduleRunResult.suites.flatMap(_.tests).map(_.status)
+    val failedTests = allTests.filter(_.isInstanceOf[org.virtuslab.ideprobe.protocol.TestStatus.Failed])
```
Comment (lukaszwawrzyk) [inline extensions/scala/tests/src/test/scala/org/virtuslab/ideprobe/scala/ScalaTestRunConfigurationTest.scala:28]: if you need to do such thing, it is probably signal that our api is lacking. You could either add testCount, failedTestCount, ignoredTestCount to the run result, or add .isFailed to test status. Or both.

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala [hunk 3]
```diff
@@ -108,6 +108,14 @@ object RunConfigurations extends IntelliJApi {
     }
   }
 
+  def runFailedTests(projectRef: ProjectRef)(implicit ec: ExecutionContext): TestsRunResult = {
```
Comment (lukaszwawrzyk) [inline probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:111]: rerunFailedTests for consistency

Comment (odisseus) [reply probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:111]: Ok

File: extensions/scala/tests/src/test/scala/org/virtuslab/ideprobe/scala/ScalaTestRunConfigurationTest.scala (hunk 2 above)
Comment (odisseus) [reply extensions/scala/tests/src/test/scala/org/virtuslab/ideprobe/scala/ScalaTestRunConfigurationTest.scala:28]: Good point, but I'm not going to do that in this PR if I don't have to add new tests.

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/Projects.scala (hunk 1 above)
Comment (odisseus) [reply probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/Projects.scala:125]: Yes, but it [happens to work anyway](https://github.com/JetBrains/intellij-community/blob/46306f2ace10a4b8e5b5416d95e3c6c9831a29fc/platform/core-api/src/com/intellij/openapi/vfs/VirtualFileManager.java#L158-L162).

**Review by lukaszwawrzyk — APPROVED** [review]

**Review by MichalBednarz — APPROVED** [review]

**Pull request merged**